packages/driver/src/cy/assertions.ts

```diff
@@ -143,9 +143,19 @@ export const create = (Cypress, cy) => {
       obj.$el = $dom.wrap(value)
     }
 
-    // if we are simply verifying the upcoming
-    // assertions then do not immediately end or snapshot
-    // else do
+    // `verifying` represents whether we're deciding whether or not to resolve
```

Comment:
maybe out of scope, but thoughts on renaming `verifying` to something a little more self-explanatory.... idea might be `isVerifyingUpcomingAssertions` (the terminology you emphasized in your log description).

Reply:
I think I'd like to save renaming things for a later date. My real goal would be to make a flowchart of the logic and consider a refactor of this area of code - Brian spoke to some regrets that we didn't just re-implement parts of Chai, rather than weaving in and out of 3rd party code and keeping track of state internally to make it behave as we want.